src/com/github/johnthagen/cppcheck/CppcheckInspection.java

```diff
@@ -172,7 +172,8 @@ private static String prependIncludeDir(@NotNull String cppcheckOptions, @NotNul
 
     private static String executeCommandOnFile(@NotNull final String command,
                                                @NotNull final String options,
-                                               @NotNull final String filePath) throws ExecutionException {
+                                               @NotNull final String filePath,
+                                               final String cppcheckMisraPath) throws ExecutionException {
```

Comment:
Could `cppcheckMisraPath` be set to `@NotNull` and checked before it enters this function? Then it would match the other `String`s that are already `@NotNull`.

Reply:
I wanted to use `StringUtils.defaultValue()` (or whatever it's called - it's been a while since I did Java), but that requires an additional dependencies. And also use this for all the properties since the other `Properties.get()` calls are also suspect to NPE.

I think this is fine for now so the plugin will work again for now.

Reply:
Strikes me as weird that, even though I followed the exact way of storing and getting the path that the options field uses, that its only now that it starts causing issues.  I'd be happy to take a deeper look in at some point if required to see why, but for the time being, this gets everything working again...

Reply:
I assume its because the property is never populated when the plugin is just updated, but never configured afterwards.